spec/mercure.md

```diff
@@ -323,127 +353,145 @@ allowed to, a 403 HTTP status code **SHOULD** be returned.
 
-If the `mercure.subscribe` array contains the reserved string value `*`, then the subscriber is
-authorized to receive updates destined for all targets.
+# Reconnection, State Reconciliation and Event Sourcing {#reconciliation}
 
-# Reconnection and State Reconciliation {#reconciliation}
+The protocol allows to reconciliate states after a reconnection. It can also be used to implement an
+[Event store](https://en.wikipedia.org/wiki/Event_store).
 
-To allow re-establishment in case of connection lost, events dispatched by the hub **SHOULD**
-include an `id` property. The value contained in this `id` property **SHOULD** be a globally unique
-identifier. To do so, a UUID [@RFC4122] **MAY** be used.
+To allow re-establishment in case of connection lost, events dispatched by the hub **MUST** include
+an `id` property. The value contained in this `id` property **SHOULD** be an IRI [@!RFC3987]. An
+UUID [@RFC4122] or a DID [@W3C.WD-did-core-20200421] **MAY** be used.
 
 According to the server-sent events specification, in case of connection
 lost the subscriber will try to automatically re-connect. During the
 re-connection, the subscriber **MUST** send the last received event id in a
 [Last-Event-ID](https://html.spec.whatwg.org/multipage/iana.html#last-event-id) HTTP header.
 
-The server-sent events specification doesn't allow this HTTP header to be set during the first
-connection (before a reconnection). In order to fetch any update dispatched between the initial
-resource generation by the publisher and the connection to the hub, the subscriber **MUST** send the
-event id provided during the discovery in the `last-event-id` link's attribute in a query parameter
-named `Last-Event-ID` when connecting to the hub. See (#discovery).
+In order to fetch any update dispatched between the initial resource generation by the publisher and
+the connection to the hub, the subscriber **MUST** send the event id provided during the discovery
+in the `last-event-id` as the last event id. See (#discovery).
+
+`EventSource` implementations may not allow to set HTTP headers during the first connection (before
+a reconnection) and implementations in web browsers don't allow to set it.
+
+To work around this problem, the hub **MUST** also allow to pass the last event id in a query
+parameter named `Last-Event-ID`.
 
 If both the `Last-Event-ID` HTTP header and the query parameter are present, the HTTP header
 **MUST** take precedence.
 
 If the `Last-Event-ID` HTTP header or query parameter exists, the hub **SHOULD** send all events
 published following the one bearing this identifier to the subscriber.
 
-The hub **MAY** discard some messages for operational reasons. The subscriber **MUST NOT** assume
-that no update will be lost, and **MUST** re-fetch the original topic to ensure this (for instance,
-after a long disconnection time).
+The reserved value `-1` can be used to hint the hub to send all updates it has for the subscribed
```

Comment:
This is confusing to me. I'd have expected `-1` to actually be used as a relative value to only return the last event. I'd go for a much more explicit `all` constant.